### ISSUE TITLE
Unable to Group and Stratify Simultaneously

### ISSUE THREAD

**Issue by DarioS** [mayer79/splitTools#32]
A user might need to group on one variable and stratify on another. For example, group by Patient ID and stratify by Therapy Response.

![Image](https://github.com/user-attachments/assets/62cc412b-aaf3-40ec-a201-7a7515f13efa)

For a particular split, training set should have cells from both classes and the cells from a patient should not appear in train set and test set.

![Image](https://github.com/user-attachments/assets/ef56e8fe-9b7f-44e8-9e9c-06a08cde0882)

**Comment by mayer79** [comment]
Fair point, but tricky. Let's see how other implementation are dealing with this first.
